RecoLocalCalo/HGCalRecAlgos/src/HGCalImagingAlgo.cc

```diff
@@ -4,49 +4,43 @@
 #include "Geometry/Records/interface/IdealGeometryRecord.h"
 #include "Geometry/CaloGeometry/interface/CaloSubdetectorGeometry.h"
 #include "Geometry/CaloGeometry/interface/CaloCellGeometry.h"
+#include "DataFormats/HcalDetId/interface/HcalSubdetector.h"
 
 #include "RecoEcal/EgammaCoreTools/interface/PositionCalc.h"
 //
 #include "DataFormats/CaloRecHit/interface/CaloID.h"
 
 
-// Create a vector of Hexels associated to one cluster from a collection of HGCalRecHits - this can be used 
-// directly to make the final cluster list - this method can be invoked multiple times for the same event 
-// with different input (reset should be called between events)
-void HGCalImagingAlgo::makeClusters(const HGCRecHitCollection& hits)
-{
-  //used for speedy search 
-  std::vector<std::vector<KDNode> >points(2*(maxlayer+1));
-  std::vector<KDTree> hit_kdtree(2*(maxlayer+1));
-
-  std::vector<std::array<float,2> > minpos(2*(maxlayer+1),{ {0.0f,0.0f} }),maxpos(2*(maxlayer+1),{ {0.0f,0.0f} });
-
-  //  std::vector<std::vector<Hexel> > points(2*(maxlayer+1)); //a vector of vectors of hexels, one for each layer
-  //@@EM todo: the number of layers should be obtained programmatically - the range is 1-n instead of 0-n-1...
-
-
-  if (verbosity < pINFO)
-    {
-      std::cout << "-------------------------------------------------------------" << std::endl;
-      std::cout << "HGC Imaging algorithm invoked for " << std::endl;
-      std::cout << "delta_c " << delta_c << " kappa " << kappa;
-      //      if( doSharing ) std::cout << " showerSigma " << std::sqrt(sigma2);
-      std::cout << std::endl;
-    }
-
+void HGCalImagingAlgo::populate(const HGCRecHitCollection& hits){
```

Comment:
In this whole class implementation, please replace cout's with thread safe MessageLogger (and remove them when commented out and not needed)